src/workspace-resolver.ts

```diff
@@ -0,0 +1,144 @@
+import path from 'node:path'
+import yaml from 'js-yaml'
+import { readFileContents } from './fs-utils'
+import { IGlobber } from './globber'
+import { ensureWorkspacesArray, uniq } from './util'
+import type { IPackageFile } from './types'
+
+/**
+ * Files service.
```

Comment:
Let's have this comment describe what the interface is for (specifically, supporting workspaces across package managers so we can sync types in all packages within the workspace)

Reply:
Silly me, I forgot to push. Updated that comment as well.